plugins/pdf-extractor/src/plugin.ts

```diff
@@ -0,0 +1,102 @@
+import axios from 'axios'
+import api, { Flatfile } from '@flatfile/api'
+import { FlatfileEvent } from '@flatfile/listener'
+import * as fs from 'fs-extra'
+import * as R from 'remeda'
+
+/**
+ * Plugin config options.
+ *
+ * @property {string} apiKey - `pdftables.com` API key
+ * @property {boolean} debug - show helpful messages useful for debugging (usage intended for development)
+ */
+export interface PluginOptions {
+  readonly apiKey: string
+  readonly debug?: boolean
+}
+
+export const run = async (
+  event: FlatfileEvent,
+  file: Flatfile.File_,
+  buffer: Buffer,
+  opts: PluginOptions
+): Promise<void> => {
+  const { environmentId, spaceId } = event.context
+
+  if (file.ext !== 'pdf' || file.mode !== 'import') {
+    return
+  }
+
+  if (R.isEmpty(opts.apiKey) && opts.debug) {
+    logWarn('Found invalid API key')
+
+    return
+  }
+
+  try {
+    const url: string = `https://pdftables.com/api?key=${opts.apiKey}&format=csv`
+    const fileName: string = `${file.name.replace(
+      '.pdf',
+      ''
+    )} (Converted PDF)-${currentEpoch()}.csv`
+
+    const formData = new FormData()
+    formData.append('file', new Blob([buffer]))
+
+    const response = await axios.postForm(url, formData)
+
+    if (response.status !== 200) {
+      logError('Failed to convert PDF on files.com')
+
+      return
+    }
+
+    fs.writeFile(fileName, response.data, async (err: unknown) => {
+      if (err && opts.debug) {
+        logError('Error writing file to disk')
+
+        return
```

Comment:
Shouldn't this `return` whether or not `opts.debug` is set?

Reply:
Yes, good catch. Thank you